### ISSUE TITLE
pre-trained vgg16 for extracting features my own dataset with keras

### ISSUE THREAD

**Issue by sheydaf** [keras-team/keras#8300]
Hi guys. I am trying to implemet pre-trained vgg16 for extracting features on keras. I have 5 classes, so I used svm for classification but my accuracy falls shrply. please let me know if anything needs to change. Any help will be appreciated and thank you in advance! This is my codes:

from keras.applications.vgg16 import VGG16
from keras.applications.vgg16 import preprocess_input
from keras.models import Model
import numpy as np
import scipy
import cv2
import glob

j=-1
X_train = np.zeros((160, 224, 224, 3), dtype='uint8')
Y_train=np.zeros((160,), dtype='float32')
         
for i in glob.glob("E:/001/train/*.jpg"):
   j = j+1
   n= cv2.imread(i)
   img = scipy.misc.imresize(n,[224,224])
   X_train[j,:,:,:] = img
  
for i in glob.glob("E:/002/train/*.jpg"):
   j = j+1
   n= cv2.imread(i)
   img = scipy.misc.imresize(n,[224,224])
   X_train[j,:,:,:] = img    
   
for i in glob.glob("E:/003/train/*.jpg"):
   j = j+1
   n= cv2.imread(i)
   img = scipy.misc.imresize(n,[224,224])
   X_train[j,:,:,:] = img        

for i in glob.glob("E:/004/train/*.jpg"):
   j = j+1
   n= cv2.imread(i)
   img = scipy.misc.imresize(n,[224,224])
   X_train[j,:,:,:] = img

for i in glob.glob("E:/005/train/*.jpg"):
   j = j+1
   n= cv2.imread(i)
   img = scipy.misc.imresize(n,[224,224])
   X_train[j,:,:,:] = img

Y_train[0:32]= 0
Y_train[32:64]= 1
Y_train[64:96]= 2
Y_train[96:128]= 3
Y_train[128:161]= 4

j=-1

Y_valid= np.zeros((40,), dtype='float32')
X_valid = np.zeros((40, 224, 224, 3), dtype='uint8')

for i in glob.glob("E:/001/test/*.jpg"):
   j = j+1
   n= cv2.imread(i)
   img = scipy.misc.imresize(n,[224,224])
   X_valid[j,:,:,:] = img
  
for i in glob.glob("E:/002test/*.jpg"):
   j = j+1
   n= cv2.imread(i)
   img = scipy.misc.imresize(n,[224,224])
   X_valid[j,:,:,:] = img    
   
for i in glob.glob("E:/003/test/*.jpg"):
   j = j+1
   n= cv2.imread(i)
   img = scipy.misc.imresize(n,[224,224])
   X_valid[j,:,:,:] = img        

for i in glob.glob("E:/004/test/*.jpg"):
   j = j+1
   n= cv2.imread(i)
   img = scipy.misc.imresize(n,[224,224])
   X_valid[j,:,:,:] = img

for i in glob.glob("E:/005/test/*.jpg"):
   j = j+1
   n= cv2.imread(i)
   img = scipy.misc.imresize(n,[224,224])
   X_valid[j,:,:,:] = img

   
   
Y_valid[0:8]= 0
Y_valid[8:16]= 1
Y_valid[16:24]= 2
Y_valid[24:32]= 3
Y_valid[32:41]= 4

base_model = VGG16(weights='imagenet')
base_model.summary()

model = Model(inputs=base_model.input, outputs=base_model.get_layer('predictions').output)
X_train = X_train.astype('float32')
X_valid = X_valid.astype('float32')
X_train = preprocess_input(X_train)
X_valid = preprocess_input(X_valid)
X_trainf = model.predict(X_train)
X_validf = model.predict(X_valid)
print("Feature vector dimensions: ",X_trainf.shape)

from sklearn import svm
#    

model1=svm.SVC(kernel='linear').fit(X_trainf, Y_train)#%--0.4249
from keras.applications.imagenet_utils import decode_predictions

fit1 = model1.fit(X_trainf, Y_train)

predictions = fit1.predict(X_validf)
pred = np.round(predictions)
pred_acc = pred == Y_valid
pred_acc1 = (np.sum(pred_acc) / np.size(Y_valid))
